### PR TITLE
Searchable Select Additional Props

### PR DESCRIPTION
**Pull request by reedsa**
Issues addressed:
* Added missing props on the main Select component, like placeholder and disabled.
* Properly call the event handler
* Fix selector issue by moving id to div container element

### CODE REVIEW

File: src/select/SearchableSelect.tsx [hunk 1]
```diff
@@ -80,8 +87,8 @@ class SearchableSelect extends Component<any> {
               tabSelectsValue={false}
             />
           </div>
-        ) : null}
-        {isOpen ? <Blanket onClick={() => this.toggleOpen()} /> : null}
+        )}
+        {isOpen && <Blanket onClick={() => this.toggleOpen()} />}
```
Comment (chrischalstrom) [inline src/select/SearchableSelect.tsx]: seems like a fragment could wrap the blanket and the input-search-wrap

File: src/select/index.tsx [hunk 2]
```diff
@@ -319,10 +322,10 @@ const Select: React.SFC<any> = props => {
         </label>
       )}
       {props.isSearchable && !props.isMulti ? (
-        <SearchableSelect {...props} />
+        <SearchableSelect {...restProps} />
       ) : (
         <ReactSelect
-          {...props}
+          {...restProps}
```
Comment (chrischalstrom) [inline src/select/index.tsx]: won't this change where the `id` was?  that might break some tests?

File: src/select/index.tsx [hunk 3]
```diff
@@ -292,6 +292,8 @@ export const DropdownIndicator: React.SFC<
 };
 
 const Select: React.SFC<any> = props => {
```
Comment (chrischalstrom) [inline src/select/index.tsx]: it is annoying that the type is `any`

File: src/select/SearchableSelect.tsx (hunk 1 above)
Comment (reedsa) [reply src/select/SearchableSelect.tsx]: Even if they are siblings along with the ReactSelect inside the wrapping div?

**Comment by thinkingserious** [comment]
Hello @reedsa,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**